### PR TITLE
feat(stryker): #438 Extensive config validation

### PR DESCRIPTION
**Pull request by hrb90**

### CODE REVIEW

File: packages/stryker/src/ConfigValidator.ts [hunk 1]
```diff
@@ -48,6 +57,27 @@ export default class ConfigValidator {
     }
   }
 
+  private validateLogLevel() {
+    const logLevel = this.strykerConfig.logLevel;
+    const VALID_LOG_LEVEL_VALUES = ['fatal', 'error', 'warn', 'info', 'debug', 'trace', 'all', 'off'];
+    if (VALID_LOG_LEVEL_VALUES.indexOf(logLevel) < 0) {
+      this.invalidate('\`logLevel\` is invalid, expected one of \`fatal\`, \`error\`, \`warn\`, \`info\`, \`debug\`, \`trace\`, \`all\` and \`off\`');
```
Comment (nicojs) [inline packages/stryker/src/ConfigValidator.ts]: You could use `VALID_LOG_LEVEL_VALUES .join(',')` here to reduce duplication.

File: packages/stryker/src/ConfigValidator.ts [hunk 2]
```diff
@@ -14,6 +14,15 @@ export default class ConfigValidator {
...
     this.validateTestFramework();
     this.validateThresholds();
+    this.validateLogLevel();
+    this.validateTimeout();
+    this.validateIsNumber('port', this.strykerConfig.port);
+    this.validateIsNumber('maxConcurrentTestRunners', this.strykerConfig.maxConcurrentTestRunners);
+    this.validateIsString('mutator', this.strykerConfig.mutator);
+    this.validateIsStringArray('plugins', this.strykerConfig.plugins);
+    this.validateIsStringArray('reporter', this.strykerConfig.reporter);
+    this.validateIsStringArray('transpilers', this.strykerConfig.transpilers);
```
Comment (nicojs) [inline packages/stryker/src/ConfigValidator.ts]: I love this, very much readable!

File: packages/stryker/src/ConfigValidator.ts (hunk 1 above)
Comment (simondel) [reply packages/stryker/src/ConfigValidator.ts]: This is actually done in `validateCoverageAnalysis`. I'm a fan on using the join here as well.

**Comment by hrb90** [comment]
@simondel I went ahead and changed as many of the error messages as possible to follow the format "Value {value} is invalid for {field}. Expected {expectation}." I left one pair of backticks in the messages (around the field) for back consistency.

**Comment by simondel** [comment]
Awesome! Looking good :) @nicojs  Shall we merge this?

**Comment by nicojs** [comment]
Great job! 🥇

**Pull request merged**